### ISSUE TITLE
New Feature Idea: Keep undefined keys in a dict

### ISSUE THREAD

**Issue by fletort** [rnag/dataclass-wizard#57]
### Description

It would be really usefull to have a Meta to can Include undefined keys in a global dict of the dataclass (as lib dataclasses-json does with its option (undefined=Undefined.INCLUDE https://lidatong.github.io/dataclasses-json/#handle-unknown-extraneous-fields-in-json).

**Comment by fletort** [comment]
( I forgot: i can contribute :-) )

**Comment by rnag** [comment]
Hi @fletort! I certainly agree with you, I feel like this would be a useful feature to have. I had a look at the docs that were linked and I liked that was one Meta config and that you could set an enum to represent the action you wanted to take when an unknown JSON field was encountered in the input data - for example raise/exclude/include.

Unfortunately I feel as it is too late to update our existing Meta field `raise_on_unknown_json_key` to be an Enum, even though that would be nice. Though I can confirm that the default behavior of the library is to *exclude* any additional JSON keys that don't map to any dataclass fields. Therefore, I do agree that the best option in this case would be to add a new flag to the `Meta` config, something like `catch_unknown_json_keys` (or include) for example. We could probably also set a default attribute on the dataclass, something like `__extra_json_data__`, if the user doesn't define a catch-all field in the dataclass. Of course the actual implementation detail might vary, and will leave it up to whatever approach can work best.

I would also certainly welcome any contribution or PR to achieve this. If are you curious, you can check out the [Contributing Guide](https://dataclass-wizard.readthedocs.io/en/latest/contributing.html) I have added in the docs. I might need to update it slightly at some point, but hopefully it should provide a good idea of how to get started.

**Comment by rnag** [comment]
@fletort Hi there 👋 , hope you're having a good day!

Sorry for the (long-ish) delay, but I just got some time to look into this and decided to implement it thoroughly. Thank you also for the linked docs, I had a look and borrowed a lot of concepts from how `dataclasses-json` had implemented it, including even the name `CatchAll` which I decided I liked so much, so I copied it verbatim 🤐 .

This new feature idea is now live as of `v0.29.0`!

Anyway, I have added [a section](https://dataclass-wizard.readthedocs.io/en/latest/index.html#save-or-catch-all-unknown-json-keys) for this in the main docs, and I even have a dedicated section on [Handling Unknown JSON Keys](https://dataclass-wizard.readthedocs.io/en/latest/common_use_cases/handling_unknown_json_keys.html) on usage of this in the docs! 🎉
